core/shared/src/main/scala/scalaz/zio/FiberRef.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalaz.zio
+
+import scalaz.zio
+
+/**
+ * Fiber's counterpart for Java's `ThreadLocal`. Value is automatically propagated
+ * to child on fork and merged back in after joining child.
+ * {{{
+ * for {
+ *   fiberRef <- FiberRef.make("Hello world!")
+ *   child <- fiberRef.set("Hi!).fork
+ *   result <- child.join
+ * } yield result
+ * }}}
+ *
+ * `result` will be equal to "Hi!" as changes done by child were merged on join.
+ *
+ * @param initial
+ * @tparam A
+ */
+final class FiberRef[A](private[zio] val initial: A) extends Serializable {
```

Comment:
Do we even need this `initial`? I am just thinking that it could be a giant data structure and we would want to allow it to be garbage collected.

So I think we can simplify to:

```scala
final class FiberRef[A] extends Serializable {
```

Reply:
This is solved in comment https://github.com/scalaz/scalaz-zio/pull/665#discussion_r285396365